components/match2/wikis/splitgate/match_group_input_custom.lua

```diff
@@ -0,0 +1,562 @@
+	if match.opl then links.opl = 'https://www.opleague.eu/match/' .. match.opl end
+	if match.esl then links.esl = 'https://play.eslgaming.com/match/' .. match.esl end
+	if match.faceit then links.faceit = 'https://www.faceit.com/en/rainbow_6/room/' .. match.faceit end
+	if match.lpl then links.lpl = 'https://letsplay.live/match/' .. match.lpl end
+	if match.r6esports then links.r6esports = 'https://www.r6esports.com.br/en/match/' .. match.r6esports end
+	if match.stats then links.stats = match.stats end
+
+	-- apply vodgames
+	for index = 1, MAX_NUM_VODGAMES do
+		local vodgame = match['vodgame' .. index]
+		if not Logic.isEmpty(vodgame) then
+			local map = match['map' .. index] or {}
+			map.vod = map.vod or vodgame
+			match['map' .. index] = map
+		end
+	end
+	return match
+end
+
+function matchFunctions.getExtraData(match)
+	match.extradata = {
+		matchsection = Variables.varDefault('matchsection'),
+		lastgame = Variables.varDefault('last_game'),
+		comment = match.comment,
+		mapveto = matchFunctions.getMapVeto(match),
+		mvp = matchFunctions.getMVP(match),
+		isconverted = 0
+	}
+	return match
+end
+
+-- Parse the mapVeto input
+function matchFunctions.getMapVeto(match)
+	if not match.vetoes then return {} end
+
+	local vetoes = mw.text.split(match.vetoes or '', ',')
+	match.vetoes = nil
+	local vetoesBy = mw.text.split(match.vetoesBy or '', ',')
+	match.vetoesBy = nil
+	local index = 1
+	local currentVetoMap = mw.text.trim(vetoes[1])
+	local vetoData = {}
+
+	while not String.isEmpty(currentVetoMap) do
+		local by = tonumber(mw.text.trim(vetoesBy[index]) or '')
+		vetoData[index] = { map = currentVetoMap, by = by }
+		index = index + 1
+		currentVetoMap = mw.text.trim(vetoes[index] or '')
+	end
+
+	return vetoData
+end
+
+-- Parse MVP input
+function matchFunctions.getMVP(match)
+	if not match.mvp then return {} end
+	local mvppoints = match.mvppoints or 1
+
+	-- Split the input
+	local players = mw.text.split(match.mvp, ',')
+
+	-- Trim the input
+	for index,player in pairs(players) do
+		players[index] = mw.text.trim(player)
+	end
+
+	return {players=players, points=mvppoints}
+end
+
+function matchFunctions.getOpponents(match)
+	-- read opponents and ignore empty ones
+	local opponents = {}
+	local isScoreSet = false
+	for opponentIndex = 1, MAX_NUM_OPPONENTS do
+		-- read opponent
+		local opponent = match['opponent' .. opponentIndex]
+		if not Logic.isEmpty(opponent) then
+			--retrieve name and icon for teams from team templates
+			if opponent.type == 'team' and
+				not Logic.isEmpty(opponent.template, match.date) then
+					local name, icon, template = opponentFunctions.getTeamNameAndIcon(opponent.template, match.date)
+					opponent.template = template or opponent.template
+					opponent.name = opponent.name or name
+					opponent.icon = opponent.icon or icon
+			end
+			if not String.isEmpty(opponent.name) then
+				opponent.name = mw.ext.TeamLiquidIntegration.resolve_redirect(opponent.name)
+			end
```

Comment:
Is this correct for teams?

Reply:
imo team name should indicate the identifier of the team (i.e. their pagename) and not displayname or the name of the template (for template we have the template field)

Reply:
OK, looked at it in it's whole and I agree with you, other than I think the parameter name (`opponent.name`) should have been called page, pagename or link instead. But I know it's the fieldnames in lpdb so can't be changed.